app/grandchallenge/reader_studies/forms.py

```diff
@@ -688,6 +691,56 @@ class Meta(PermissionRequestUpdateForm.Meta):
         model = ReaderStudyPermissionRequest
 
 
+class GroundTruthFromAnswersForm(SaveFormInitMixin, Form):
+    user = ModelChoiceField(
+        queryset=get_user_model().objects.none(),
+        required=True,
+        help_text=format_html(
+            "Select a user whose answers will be consumed. "
+            "Only users that have <strong>completed the reader study</strong> are valid options."
+        ),
+        widget=Select2Widget,
+    )
+
+    def __init__(self, *args, reader_study, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self._reader_study = reader_study
+
+        self.fields["user"].queryset = (
+            get_user_model()
+            .objects.filter(answer__question__reader_study=reader_study)
+            .distinct()
+        )
+
+    def clean(self):
+        if self._reader_study.has_ground_truth:
+            raise ValidationError(
+                "Reader study already has ground truth. Ground truth cannot be updated. "
+                "Please, first delete the ground truth."
+            )
+        return super().clean()
+
+    def clean_user(self):
+        user = self.cleaned_data["user"]
+
+        progress = self._reader_study.get_progress_for_user(user)
+        if progress["questions"] != 100.0:
+            raise ValidationError("User has not completed the reader study!")
+
+        return user
+
+    def create_ground_truth(self):
+        answers = Answer.objects.filter(
+            question__in=self._reader_study.ground_truth_applicable_questions,
+            creator=self.cleaned_data["user"],
+        )
+        answers.update(is_ground_truth=True, score=None)
```

Comment:
Should all answers for the reader study have their scores reset?

Reply:
It is intentional: it is a clear signal that all scores are invalid until we can get to re-calculating them.

Reply:
In practice that is the case: when deleting the ground-truth all scores are reset and a user (should) not get here without deleting the ground-truth.

I could play it super safe and have it reset all scores here? It is only one additional query to the DB after all?